File: Lesson05/Intents/app/src/main/java/com/codingblocks/intents/MainActivity.java
package com.codingblocks.intents;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void newActivity(View view) {
        //What if I want to start the SecondActivity on clicking this button?

        Intent startIntent = new Intent(getBaseContext(),
                SecondActivity.class);

        EditText input = findViewById(R.id.etName);
        String inputString = input.getText().toString();

        startIntent.putExtra("a", inputString);

//        getBaseContext().getColor(R.color.colorPrimary);
//        getBaseContext().getString(R.string.test);
//        getBaseContext().getDrawable(R.drawable.ic_launcher_background);

//        Log.e("TAG",
//                getPackageName() + "" +
//
//                        SecondActivity.class.getName());

        startActivity(startIntent);

    }
}

File: Lesson15/Parcelable/app/src/main/java/com/codingblocks/parcelable/MainActivity.kt
package com.codingblocks.parcelable

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {

    val note = Note("test", "test", 1, 11)

    val notesList = arrayListOf(note, note, note, note)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val intent = Intent(this, NewActivity::class.java)

//        val gson = Gson()
//        val noteString = gson.toJson(note)
//        intent.putExtra("NOTE",noteString)

        intent.putExtra("NOTE", note)
        intent.putParcelableArrayListExtra("NOTES", notesList)

        //Serialization and Deserialization

        startActivity(intent)
    }
}
